test/unit/matrix/test_matrix.cpp

```diff
@@ -1422,15 +1575,38 @@ TEST(MatrixExceptionPropagation, RWPropagatesInReadAccess) {
   EXPECT_THROW(f.get(), CustomException);
 }
 
+TEST(MatrixExceptionPropagation, RWPropagatesInReadAccessWithSenderAdaptors) {
+  namespace ex = pika::execution::experimental;
+
+  auto matrix = createMatrix<TypeParam>();
+
+  auto s = matrix.readwrite_sender(LocalTileIndex(0, 0)) |
+           then(unwrapping([](auto&&) { throw CustomException{}; })) | ensure_started();
```

Comment:
super minor: probably we could take the chance to refactor also this 😛 your choice 😄

Reply:
Sorry, nope :stuck_out_tongue: won't do that here.